src/installer.ts

```diff
@@ -297,24 +337,36 @@ export class Installer extends EventEmitter {
     if (installedVersion === version) {
       d(`already installed`);
     } else {
-      const zipFile = await this.ensureDownloaded(version, opts);
-      this.setState(version, InstallState.installing);
-      d(`installing from "${zipFile}"`);
-      await fs.emptyDir(electronInstall);
-      // FIXME(anyone) is there a less awful way to wrangle asar
-      // @ts-ignore: yes, I know noAsar isn't defined in process
-      const { noAsar } = process;
-      try {
-        // @ts-ignore: yes, I know noAsar isn't defined in process
-        process.noAsar = true;
-        await extract(zipFile, { dir: electronInstall });
-      } finally {
-        // @ts-ignore: yes, I know noAsar isn't defined in process
-        process.noAsar = noAsar; // eslint-disable-line
+      const { path: zipFile, preInstalled: isAlreadyInstalled } =
```

Comment:
nit(non-blocking): it's a bit confusing that `isAlreadyInstalled` is in a different control flow branch from the `already installed` debug statement.

Would `alreadyExtracted` be clearer?

Reply:
Yup, But now that I think of - maybe `alreadyExtracted` goes better in the `ElectronBinary` interface than `preInstalled`.